fastlane_core/lib/fastlane_core/project.rb

```diff
@@ -288,6 +292,9 @@ def build_settings(key: nil, optional: true)
           timeout = FastlaneCore::Project.xcode_build_settings_timeout
           retries = FastlaneCore::Project.xcode_build_settings_retries
           @build_settings = FastlaneCore::Project.run_command(command, timeout: timeout, retries: retries, print: !self.xcodebuild_list_silent)
+          if @build_settings.empty?
+            UI.error("Could not read build settings. Make sure that the scheme \"#{options[:scheme]}\" is configured for running by going to Product → Scheme → Edit Scheme…, selecting the \"Build\" section, checking the \"Run\" checkbox and closing the scheme window.")
```

Comment:
I’m not 100% sure. I chose to continue because I only tested in the context of `scan`. I don’t want to break other tools since project.rb may be used everywhere.